jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -0,0 +1,204 @@
+    }
+    will be assigned to the state.
+
+    @param {String} markdown The raw markdown.
+    @return {Object} Parsed metadata.
+  */
+  parseMarkdownMetadata(markdown) {
+    let parsedMetadata = {};
+    if (markdown.indexOf('---') === 0) {
+      let lineByLine = markdown.split('\n');
+      let metadata = [];
+      lineByLine.shift();
+      while(lineByLine[0] !== '---') {
+        metadata.push(lineByLine.shift());
+      }
+      lineByLine.shift();
+
+      metadata.forEach(option => {
+        const splitOption = option.split(':');
+        return parsedMetadata[splitOption[0].trim()] = splitOption[1].trim();
+      });
+
+      return parsedMetadata;
+    }
+  }
+
+  /**
+    Replace variables within the markdown with a pre-defined list.
+
+    @param {String} markdown The raw markdown.
+    @return {String} The replaced string.
+  */
+  replaceVars(markdown) {
+    let replaced = markdown;
+
+    Object.keys(this.variables).forEach(variable => {
+      replaced = replaced.split(`{${variable}}`).join(this.variables[variable].content);
+    });
+
+    return replaced;
+  }
+
+  /**
+    Catches click events on the contents of the modal. This allows replaced
+    variables to have React functionality.
+
+    @param {SyntheticEvent} evt The click event.
+  */
+  _handleContentClick(evt) {
+    const variable = evt.target.getAttribute('data-variable');
+    if (variable) {
+      const resolvedVar = this.variables[variable];
+      if (resolvedVar && resolvedVar.onClick) {
+        resolvedVar.onClick.call(this);
+      }
+    }
+  }
+
+  /**
+    Show the details page of a charm or bundle based on the click.
+
+    @param {SyntheticEvent} evt The click event.
+  */
+  _handleViewDetails(evt) {
+    if (evt) evt.preventDefalt();
+    const storeState = {
+      profile: null,
+      search: null,
+      store: this.props.entityUrl.path()
+    };
+
+    this.props.changeState(storeState);
+  }
+
+  render() {
+    let classes = [
+      'modal--right',
+      'modal--auto-height',
+      'post-deployment'
+    ];
+    if (this.state.content) {
+      return (
+        <juju.components.Modal
+          closeModal={this.props.closePostDeployment}
+          extraClasses={classes.join(' ')}>
+          <div onClick={this._handleContentClick.bind(this)}
+            dangerouslySetInnerHTML={{__html: this.state.content}} />
+        </juju.components.Modal>
+      );
+    } else if (this.props.displayName) {
+      classes.push('post-deployment--simple');
+      return (
+        <juju.components.Modal
+          closeModal={this.props.closePostDeployment}
+          extraClasses={classes.join(' ')}>
+          <p>
+            {this.props.displayName}
+          &nbsp;</p>
+          <span
+            role="button"
+            className="link"
+            onClick={this._handleViewDetails.bind(this)}>View details</span>
+        </juju.components.Modal>
+      );
+    }
+    return null;
+  }
+}
+
+PostDeployment.propTypes = {
```

Comment:
It would be nice to shape up this properties, like entity url, displayname, files and getFile could be grouped together as entityAPI (url, name, files, getFile). Also changeState and closePostDeployment could be grouped.
This PR is already big, so take it or leave it, really.